### PR TITLE
MINOR: add session windows doc to streams.html

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@miguno this is largely c&p from what you wrote. What else do we want to add her?

**Comment by miguno** [comment]
@dguy: What about diagrams?

**Comment by miguno** [comment]
We should also consider adding code snippets.

**Comment by dguy** [comment]
@guozhangwang @mjsax @enothereska - i spoke to @miguno offline and he was happy to merge this as is, but... there is a bigger question w.r.t updating all the windowing docs here. Not sure what the plan is.

**Comment by enothereska** [comment]
Ok, LGTM as is too.

File: docs/streams.html [hunk 1]
```diff
@@ -539,6 +539,12 @@
...
         <li><b>Tumbling time windows</b> are a special case of hopping time windows and, like the latter, are windows based on time intervals. They model fixed-size, non-overlapping, gap-less windows. A tumbling window is defined by a single property: the window's size. A tumbling window is a hopping window whose window size is equal to its advance interval. Since tumbling windows never overlap, a data record will belong to one and only one window.</li>
         <li><b>Sliding windows</b> model a fixed-size window that slides continuously over the time axis; here, two data records are said to be included in the same window if the difference of their timestamps is within the window size. Thus, sliding windows are not aligned to the epoch, but on the data record timestamps. In Kafka Streams, sliding windows are used only for join operations, and can be specified through the <code>JoinWindows</code> class.</li>
+        <li><b>Session windows</b> Session windows are used to aggregate key-based events into sessions.
```
Comment (mjsax) [inline docs/streams.html]: `<b>Session windows</b> Session windows`
don't repeat

File: docs/streams.html [hunk 2]
```diff
@@ -539,6 +539,12 @@
...
+            Sessions represent a period of activity separated by a defined gap of inactivity.
+            Any events processed that fall within the inactivity gap of any existing sessions are merged into the existing sessions.
+            If the event falls outside of the session gap, then a new session will be created.
+            Session windows are tracked independently across keys (e.g. windows of different keys typically have different start and end times) and their sizes sizes vary (even windows for the same key typically have different sizes); as such session windows can't be pre-computed and are instead derived from analyzing the timestamps of the data records.
```
Comment (mjsax) [inline docs/streams.html]: Nit: line too long

File: docs/streams.html [hunk 3]
```diff
@@ -539,6 +539,12 @@
...
         <li><b>Sliding windows</b> model a fixed-size window that slides continuously over the time axis; here, two data records are said to be included in the same window if the difference of their timestamps is within the window size. Thus, sliding windows are not aligned to the epoch, but on the data record timestamps. In Kafka Streams, sliding windows are used only for join operations, and can be specified through the <code>JoinWindows</code> class.</li>
+        <li><b>Session windows</b> Session windows are used to aggregate key-based events into sessions.
+            Sessions represent a period of activity separated by a defined gap of inactivity.
+            Any events processed that fall within the inactivity gap of any existing sessions are merged into the existing sessions.
```
Comment (mjsax) [inline docs/streams.html]: Maybe mention something like "session detection"

**Comment by mjsax** [comment]
In general, I would merge and do another PR on top... Short turn-around time FTW!

**Review by mjsax — APPROVED** [review]
LGTM

**Review by guozhangwang — APPROVED** [review]
Merged to trunk and 0.10.2.

**Pull request closed without merging**